include/gatekeeper_gk.h

```diff
@@ -99,17 +99,26 @@ struct neighbor_hash_table {
 	/* The tables that store the Ethernet headers. */
 	struct ether_cache *cache_tbl;
 
-	/* TODO Use a spin lock to edit the hash table. */
 	struct rte_hash *hash_table;
 };
 
 /* The gk forward information base (fib). */
 struct gk_fib {
 
+	/* Use a spin lock to edit the FIB table. */
+	rte_spinlock_t lock;
```

Comment:
Just to confirm, actually we can just use a single lock to synchronize the behavior of multiple writers, so that we can only allow one writer to change a FIB entry. Since the write operations are less frequent, it won't affect the performance too much. With this, as suggested above, the readers should have no issues for reading the FIB entries. Is this right? Sorry for this inconvenience, I should take notes about our discussion result then.